Plugins/Flow.Launcher.Plugin.Explorer/ViewModels/SettingsViewModel.cs

```diff
@@ -525,6 +525,16 @@ public string ExcludedFileTypes
             }
         }
 
+        public int MaxResult
+        {
+            get => Settings.MaxResult;
+            set
+            {
+                Settings.MaxResult = Math.Clamp(value, 0, 10000);
```

Comment:
You already placed min and max restrictions on the UI, is this still needed?

Reply:
In my view, real validation/restriction belongs in lower levels like viewmodel or business layer. Redundant validation on UI should be for user convenience. Of course there's always risk of two going out of sync and producing a wonky experience, so, yes, it's good to have a single source of truth.

In any case, I removed the min/max from UI so the only restriction is here now. I had to add a slight binding delay though, because otherwise WPF binding would randomly fail to bind the "corrected" value back into the UI.

Reply:
>In my view, real validation/restriction belongs in lower levels like viewmodel or business layer. Redundant validation on UI should be for user convenience. Of course there's always risk of two going out of sync and producing a wonky experience, so, yes, it's good to have a single source of truth.

Yes agree. You can put the min max as view model class variables so will keep single source of truth and remove the need to hardcode.

Reply:
Converted hardcoded limits to variables but I'm not using them in codebehind. See my other comment.